util/middleware.js

```diff
@@ -10,39 +10,42 @@ export function getUserUrl (type) {
 
 /**
  * This convenience method wraps a fetch call to the specified URL
- * with the token and api key added (if provided) to the HTTP request header.
+ * with the auth0 object and api key added (if provided) to the HTTP request header.
  * @param {string} url The URL to call.
- * @param {string} accessToken If non-null, the Authorization token to add to request header.
+ * @param {string} auth0 the auth0 object used to obtain an accessToken for secure APIs.
  * @param {string} apiKey If non-null, the API key to add to the Authorization header.
  * @param {string} method The HTTP method to execute.
  * @param {*} options Extra fetch options to pass to fetch.
  */
-export async function secureFetch (url, accessToken, method = 'get', options = {}) {
-  const headers = {
-    Authorization: `Bearer ${accessToken}`
-  }
-  if (process.env.API_KEY) headers['x-api-key'] = process.env.API_KEY
-  const res = await fetch(url, {
+async function secureFetchCore (url, auth0, method = 'get', options = {}) {
+  const { getAccessTokenSilently } = auth0
+  const accessToken = await getAccessTokenSilently()
+  // TODO: Handle errors fetching tokens.
```

Comment:
Please implement some handlers to gracefully fail when the above operation fails.

Reply:
@evansiroky Is the approach from https://github.com/ibi-group/otp-admin-ui/pull/33/commits/c98bbd30c91aef51a1e9ff0217da8bbfbd8ce50d what you had in mind?

Reply:
Partially, but mimicking a bad HTTP response seems too hacky. Since this method is only ever used within the `secureFetch` or `secureFetchHandleErrors` methods, it seems there shouldn't even be a try/catch block within `secureFetchCore` except to parse the contents of the getAccessTokenSilently error to provide a better error to the other secureFetch methods. Instead, one would expect the error to be handled with `secureFetchHandleErrors` and then not all with the `secureFetch` method.

Reply:
It sounds like we may need to remove the `secureFetch` function that doesn't check for errors, right?

Reply:
I ended up merging the two `secureFetch` methods in https://github.com/ibi-group/otp-admin-ui/pull/33/commits/496eab303ef131d2690db41da94786845d2ee264 and adjusting the SWR calls around that.

Reply:
Looks much better